src/AbTesting.php

```diff
@@ -61,41 +64,56 @@ protected function start()
     }
 
     /**
-     * Triggers a new visitor. Picks a new experiment and saves it to the session.
+     * Resets the visitor data.
+     *
+     * @return void
+     */
+    public function resetVisitor()
+    {
+        session()->flush();
+        $this->visitor = null;
+    }
+
+    /**
+     * Triggers a new visitor. Picks a new experiment and saves it to the Visitor.
+     *
+     * @param int $visitor_id An optional visitor identifier
      *
      * @return \Ben182\AbTesting\Models\Experiment|void
      */
-    public function pageView()
+    public function pageView($visitor_id = null)
     {
-        if (! session(self::SESSION_KEY_EXPERIMENT)) {
+        $visitor = $this->getVisitor($visitor_id);
+
+        if (! session(self::SESSION_KEY_GOALS)) {
```

Comment:
This is why the user gets another experiment even if he already had one

Reply:
Thanks for the pointer, moved the logic of calling start() and setting up new experiment into different logic conditions.